dm_cli/utils/resolve_local_ids.py

```diff
@@ -0,0 +1,136 @@
+import re
+
+
+def search_in_list(
+    document: list, pattern: str, target_attr: str, path: str, parent_was_dict: bool, targets: dict, references: dict
+):
+    attribute_types = map(lambda attr: (attr, type(attr)), document)
+    for i, at in enumerate(attribute_types):
+        a = at[0]
+        t = at[1]
+        if is_primitive(t):
+            search_in_primitive(a, pattern, f"{path}[{i}]", references)
+        else:
+            action = type_to_action[t]
+            action(a, pattern, target_attr, f"{path}[{i}]", False, targets, references)
+
+
+def search_in_primitive(val, pattern: str, path: str, references: dict):
+    match = re.match(pattern, str(val))
+    if match:
+        references[path] = match.group(1)
+
+
+def is_primitive(t: type):
+    return t == str or t == int or t == float or t == bool
+
+
+def is_list(t: type):
+    return t == list
+
+
+def is_dict(t: type):
+    return t == dict
+
+
+def search_in_dict(
```

Comment:
Could we add a unit test for `search_in_dict`? 

Easier to see what's going on then :)